src/tests/ftest/util/test_utils_pool.py

```diff
@@ -760,10 +700,35 @@ def set_query_data(self):
 
         Only supported with the dmg control method.
         """
-        self.query_data = []
+        self.query_data = {}
         if self.pool:
             if self.dmg:
-                self.query_data = self.dmg.pool_query(self.pool.get_uuid_str())
+                uuid = self.pool.get_uuid_str()
+                end_time = None
+                if self.pool_query_timeout.value is not None:
+                    self.log.info(
+                        "Waiting for pool %s query to be responsive with a %s "
+                        "second timeout", uuid, self.pool_query_timeout.value)
+                    end_time = time() + self.pool_query_timeout.value
+                while True:
+                    try:
+                        self.query_data = self.dmg.pool_query(uuid)
+                        break
+                    except CommandFailure as error:
+                        if end_time is not None:
+                            self.log.info(
+                                "Pool %s query still non-responsive: %s",
+                                uuid, str(error))
+                            if time() > end_time:
+                                raise CommandFailure(
+                                    "TIMEOUT detected after {} seconds while "
+                                    "waiting for pool {} query response. This "
+                                    "timeout can be adjusted via the "
+                                    "'pool/pool_query_timeout' test yaml "
+                                    "parameter.".format(
+                                        uuid, self.pool_query_timeout.value))
+                        else:
+                            raise CommandFailure(error)
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword